### PR TITLE
[custom-builds-example] [ENG-11196] Add Slacking example

### PR DESCRIPTION
**Pull request by radoslawkrzemien**
Added an example workflow using the `eas/send_slack_message` function to send messages over Slack

See: https://linear.app/expo/issue/ENG-11196/create-custom-build-example-slacking-team-members

### Config
Screenshot 2024-02-09 at 16 26 09

### Build
Screenshot 2024-02-09 at 16 25 45

### Messages
Screenshot 2024-02-09 at 16 25 37

### CODE REVIEW

File: .eas/build/send-slack-message.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+build:
+  name: Demo
```
Comment (sjchmiela) [inline .eas/build/send-slack-message.yml]:
```suggestion
  name: Slack your team when build completes
```

File: .eas/build/send-slack-message.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  steps:
+    - eas/checkout
```
Comment (sjchmiela) [inline .eas/build/send-slack-message.yml]:
```suggestion
    # For the purpose of the example-test we don't need to do the actual build.
    # - eas/build
```

File: .eas/build/send-slack-message.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    - eas/send_slack_message:
+        name: Send Slack message to a given webhook URL
```
Comment (sjchmiela) [inline .eas/build/send-slack-message.yml:8]:
```suggestion
        if: ${ failed() }
        name: Send Slack message when the build fails
```

File: .eas/build/send-slack-message.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        inputs:
+          message: 'This is a test message to plain input URL'
+          slack_hook_url: 'https://hooks.slack.com/services/[rest_of_hook_url]'
+
+    - eas/send_slack_message:
+        name: Send Slack message to a default webhook URL from SLACK_HOOK_URL secret
```
Comment (sjchmiela) [inline .eas/build/send-slack-message.yml:14]:
```suggestion
        if: ${ success() }
        name: Send Slack message to a default webhook URL from SLACK_HOOK_URL secret only when the build succeeds
```

**Review by sjchmiela — APPROVED** [review]
Can we also add a GitHub trigger so it runs on every commit to main?

File: .eas/build/send-slack-message.yml (hunk 3 above)
Comment (radoslawkrzemien) [reply .eas/build/send-slack-message.yml:8]: Added as a new step

File: .eas/build/send-slack-message.yml (hunk 4 above)
Comment (radoslawkrzemien) [reply .eas/build/send-slack-message.yml:14]: Added as a new step

**Review by szdziedzic — APPROVED** [review]

File: .eas/build/send-slack-message.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        inputs:
+          message: 'This is a test message to default URL from SLACK_HOOK_URL secret'
+
+    - eas/send_slack_message:
+        name: Send Slack message to a webhook URL from specified secret
+        inputs:
+          message: 'This is a test message to an URL from specified secret'
```
Comment (sjchmiela) [inline .eas/build/send-slack-message.yml]: nit:
```suggestion
          message: 'This is a test message to a URL from specified secret'
```
Zrzut ekranu 2024-02-12 o 13 06 04

File: .eas/build/send-slack-message.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  name: Slack your team when build completes
+  steps:
+    # For the purpose of the example-test we don't need to do the actual build.
+    # - eas/build
+
+    - eas/send_slack_message:
+        name: Send Slack message to a given webhook URL
+        inputs:
+          message: 'This is a test message to plain input URL'
+          slack_hook_url: 'https://hooks.slack.com/services/[rest_of_hook_url]'
```
Comment (sjchmiela) [inline .eas/build/send-slack-message.yml:11]: Actually… will this fail the job (Slack responding with 400 or something) or will we let the rest of the workflow pass? If it fails maybe we want to comment it out not to mark the job as failed (nor skip the rest of the workflow)?

**Review by sjchmiela — APPROVED** [review]

File: .eas/build/send-slack-message.yml (hunk 5 above)
Comment (radoslawkrzemien) [reply .eas/build/send-slack-message.yml]: Grammar nazis everywhere

File: .eas/build/send-slack-message.yml (hunk 6 above)
Comment (radoslawkrzemien) [reply .eas/build/send-slack-message.yml:11]: I don't see why would that be a problem - it's an example workflow, and the url obviously needs to be updated with the actual value. Unless I'm missing something?

Comment (sjchmiela) [reply .eas/build/send-slack-message.yml:11]: Ah, sure, [I was thinking](https://github.com/expo/eas-custom-builds-example/pull/15#pullrequestreview-1872499717) we could enable a Github trigger for this workflow and get messages in #turtle-bots every time we commit to `main` here. We can not do it too.

Comment (radoslawkrzemien) [reply .eas/build/send-slack-message.yml:11]: Ah, gotcha. I'll look into that

**Pull request merged**